### ISSUE TITLE
how to navigate to another fragment after clicking notification

### ISSUE THREAD

**Issue by razaghimahdi** [pusheco/pushe-android-studio-sample#57]
how can be able to navigate to specific fragment and destroyed previous fragments by clicking notification, I tried too many ways but it seems nothing works for that,
 I have a SplashFragment which is startDestination of navigation.xml, but when notification clicked I want to another fragment comes up,
usually one way always work but it seems doesnt work for Pushe.

for example:
MainActivity:
```kotlin


    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        navigateToSearchFragmentIfNeeded(intent)
    }

    private fun navigateToSearchFragmentIfNeeded(intent: Intent?){
        if (intent?.action==ACTION_SHOW_SEARCH_FRAGMENT){
            navHostFragment.findNavController().navigate(R.id.action_global_searchFragment)
        }
    }
```

before all of this, ACTION_SHOW_SEARCH_FRAGMENT  should set as intent value of notification.
for example:

```kotlin
 private fun getMainActivityPendingIntent() = PendingIntent.getActivity(
            this,
            0,
            Intent(this, MainActivity::class.java).also {
                it.action = ACTION_SHOW_SEARCH_FRAGMENT
            },
            FLAG_UPDATE_CURRENT
    )

val notificationBuilder = NotificationCompat.Builder(this, NOTIFICATION_CHANNEL_ID)
                .setAutoCancel(false)
                .setOngoing(true) 
                .setContentTitle("Title") 
                .setContentIntent(getMainActivityPendingIntent())
```

but there is no such things in Pushe
 
so how is that going to happen in Pushe?

or maybe another solution?

**Comment by mahdi-malv** [comment]
Hey

Didn't get the exact situation. Are you trying this when the app is open?

**Comment by razaghimahdi** [comment]
here is the situation, my whole app  is single activity(for now); also I use navigation component to handle fragment stuff, so normally app launch from Fragment-A( just a example name) but when app get start from notification clicking I want app launchs from fragment-D, i tried many ways to handle that but clicking notification shows weird behaves, finally i managed it somehow which is not efficient way to handle that;
here is what i did, so i made a class and called it NotificationManager and inside it i defined some value as LiveData to hold value of notification(custom-content) and inject it to viewModels, it is not what i want but it works  somehow .
so have any better solution?

**Comment by mahdi-malv** [comment]
Here's a solution to send some data along with your notification payload

> NOTE: Only usable for RestAPI sending - Pushe console does not support this yet

- When sending a notification choose `actionType='T'` as action type (More about actions [here](https://docs.pushe.co/docs/mobile-api/notification-actions#%D8%A8%D8%A7%D8%B2-%DA%A9%D8%B1%D8%AF%D9%86-%D8%A7%DA%A9%D8%AA%DB%8C%D9%88%DB%8C%D8%AA%DB%8C))
- set `pushe_activity_extra` value with a custom value you need along with `action_data=com.example.YouActivity` which is your activity package name
- When activity is opened (or created) check `getStringExtra` of `getIntent()` with the key `"pushe_data"`
- If such value existed, it activity was opened using notification action and payload exists

To sum up:

```js
{
"data": { 
  // ... rest of payload data such as title, content, etc.
  "action": {
    "action_type": "T",
    "action_data": "نام اکتیویتی",
    "pushe_activity_extra": "MY_CRAP"
    }  
  }
}
```

And `activity#onCreate`

```kotlin
// onCreate() {
val data = intent.getStringExtra("pushe_data") ?: ""
// check data
// }
```
